public/hist_nd.pro

```diff
@@ -158,7 +158,13 @@ function hist_nd,V,bs,MIN=mn,MAX=mx,NBINS=nbins,REVERSE_INDICES=ri
         nbins=long(nbins)       ;No fractional bins, please
         bs=float(mx-mn)/nbins   ;a correct formulation
      endif else message,'Must pass either binsize or NBINS'
-  endif else nbins=long((mx-mn)/bs+1) 
+  endif $
+  else begin
+     if n_elements(nbins) ne 0 then $
+        message, 'Overriding given NBINS (BINSIZE takes precedence)', $
```

Comment:
Don't need this message. You can put this in the docs. The "IDL way" is to just quietly do the right thing or throw an error.

Reply:
Agreed